common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -34,6 +34,9 @@ def connect_to_mongodb(
         # No 'replicaSet' in kwargs - so no secondary reads.
         mongo_client_class = pymongo.MongoClient
 
+    # If the MongoDB server uses a separate authentication database that should be specified here
+    auth_source = kwargs.pop('auth_source') or None
```

Comment:
@xitij2000, can you make this change and the other similar change in the upstream PR as well?